kits/arm/ex_gravity_compensation.cpp

```diff
@@ -8,40 +8,59 @@
  * demo.
  */
 
-// #include "lookup.hpp"
-// #include "group.hpp"
 #include "group_command.hpp"
 #include "group_feedback.hpp"
 #include "robot_model.hpp"
 #include "arm/arm.hpp"
+#include "util/mobile_io.hpp"
 #include <chrono>
+#include <thread>
+#include "hebi_util.hpp"
 
 using namespace hebi;
 using namespace experimental;
 
 int main(int argc, char* argv[])
 {
   //////////////////////////
-  ///// Arm Setup //////////
+  ///// Config Setup ///////
   //////////////////////////
 
-  arm::Arm::Params params;
+  // Config file path
+  const std::string example_config_file = "config/ex_gravity_compensation.cfg.yaml";
+  std::vector<std::string> errors;
+  
+  // Load the config 
+  const auto example_config = RobotConfig::loadConfig(example_config_file, errors);
+  for (const auto& error : errors) {
+    std::cerr << error << std::endl;
+  }
+  if (!example_config) {
+    std::cerr << "Failed to load configuration from: " << example_config_file << std::endl;
+    return -1;
+  }
 
-  // Setup Module Family and Module Names
-  params.families_ = {"Arm"}; //[change back]
-  params.names_ = {"J1_base", "J2_shoulder", "J3_elbow", "J4_wrist1", "J5_wrist2", "J6_wrist3"};
+  // For this demo, we need just the arm
+  std::unique_ptr<hebi::experimental::arm::Arm> arm;
+
+  //////////////////////////
+  ///// Arm Setup //////////
+  //////////////////////////
 
-  // Read HRDF file to seutp a RobotModel object for the 6-DoF Arm
-  params.hrdf_file_ = "kits/arm/hrdf/A-2085-06.hrdf";
+  // Create the arm object from the configuration
+  arm = hebi::experimental::arm::Arm::create(*example_config);
 
-  // Create the Arm Object
-  auto arm = arm::Arm::create(params);
+  // Keep retrying if arm not found
   while (!arm) {
-    arm = arm::Arm::create(params);
-  }
+      std::cerr << "Failed to create arm, retrying..." << std::endl;
 
-  // Load the gains file that is approriate to the arm
-  arm -> loadGains("kits/arm/gains/A-2085-06.xml");
+      // Wait for 1 second before retrying
+      std::this_thread::sleep_for(std::chrono::seconds(1));  
+
+      // Retry
+      arm = hebi::experimental::arm::Arm::create(*example_config);
```

Comment:
Again, I'd probably change back to `arm::Arm::create` instead (here and in other files :) ), since we're planning on making this non-experimental soon, and the examples will then "just work".